controllers/openshift_controller.go

```diff
@@ -419,6 +536,19 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	ds := r.processDaemonsetForMonitor()
+	if ds == nil {
+		r.Log.Error(err, "error deleting monitor Daemonset")
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 15}, nil
+	}
+	err = r.Client.Delete(context.TODO(), ds)
+	if err != nil && k8serrors.IsNotFound(err) {
```

Comment:
MInor nit: you could test `err != nil` once and have the second test `k8serrors.isNotFound(err)` inside that if.

Reply:
I have done just that :-)